### ISSUE TITLE
Automated Deployments

### ISSUE THREAD

**Issue by rachelwhitton** [pantheon-systems/documentation#1186]
## Open Questions
- Should it build/deploy branches without open Pull Requests? 
- Should accessibility cause failure? (a11y)
- Should we change the deployed branch from master? Would allow for merging external PR's prior to publishing.

## Tasks

- [x] Establish authentication methods for required services: CircleCi, Terminus, and the Pantheon Dashboard (Standardize with related task: Test Automation)
- [x] Determine when to run test suite and establish feature strategy
Using CircleCi, Terminus, and Bash, create an automation workflow for deploys:

**On PR**
- [x] Normalize the branch name (with CircleCI's [environment variables](https://circleci.com/docs/environment-variables) and GH Pull Number) to account for Multidev requirements.
- [x] Create Multidev Environment and rsync docs site
- [x] Explore scripting PR comments which include the Multidev URL 
- [x] If additional commits are pushed to the PR after it has been opened, the site should be re-generated and rsync'd to the environment.

**On Commit to Default Branch**
- [x] Given successful status, deploy to production on every commit/merge.
- [ ] Evaluate [protecting](https://github.com/blog/2051-protected-branches-and-required-status-checks) the default branch with required status checks
- [ ] Determine handling for commits directly to the master branch - consider [Enforcing a User-Based ACL System](https://git-scm.com/book/en/v2/Customizing-Git-An-Example-Git-Enforced-Policy)
- [x] Delete multidev when the branch is merged into master
